emcc.py

```diff
@@ -2074,6 +2074,14 @@ def default_setting(name, new_default):
   else:
     settings.JS_LIBRARIES.append((0, 'library_pthread_stub.js'))
 
+  if (
+      "$formatException" in settings.DEFAULT_LIBRARY_FUNCS_TO_INCLUDE or
+      settings.INCLUDE_FULL_LIBRARY and not settings.DISABLE_EXCEPTION_CATCHING
```

Comment:
Is the second clause here needed?  What test fails if you remove it?

Reply:
Without it, `test_full_js_library_except` fails with:
```
error: undefined symbol: emscripten_format_exception 
(referenced by $formatException__deps: ['emscripten_format_exception','free'], referenced by top-level compiled C/C++ code)
```

Reply:
Ok, how about just this for now:

```
# TODO: Move this into the library JS file once it becomes possible.
# See https://github.com/emscripten-core/emscripten/pull/15982
if settings.INCLUDE_FULL_LIBRARY and not settings.DISABLE_EXCEPTION_CATCHING:
   settings.EXPORTED_FUNCTIONS += ['_emscripten_format_exception', '_free']
```

Reply:
Okay I put that in and updated the test.